cajita/unit_test/tstSparseLocalGrid.hpp

```diff
@@ -0,0 +1,489 @@
+                   owned_cell_space.min( Dim::J ) );
+        EXPECT_EQ( ghosted_shared_tile_space.max( Dim::J ) *
+                       cell_num_per_tile_dim,
+                   owned_cell_space.max( Dim::J ) );
+        EXPECT_EQ( ghosted_shared_tile_space.min( Dim::K ) *
+                       cell_num_per_tile_dim,
+                   owned_cell_space.min( Dim::K ) );
+        EXPECT_EQ( ghosted_shared_tile_space.max( Dim::K ) *
+                       cell_num_per_tile_dim,
+                   owned_cell_space.max( Dim::K ) );
+    }
+    if ( cart_rank[Dim::J] + 1 < cart_dims[Dim::J] )
+    {
+        auto ghosted_shared_tile_space =
+            local_grid_ptr->sharedTileIndexSpace<2>( Ghost(), Cell(), 0, 1, 0 );
+        EXPECT_EQ( ghosted_shared_tile_space.min( Dim::I ) *
+                       cell_num_per_tile_dim,
+                   owned_cell_space.min( Dim::I ) );
+        EXPECT_EQ( ghosted_shared_tile_space.max( Dim::I ) *
+                       cell_num_per_tile_dim,
+                   owned_cell_space.max( Dim::I ) );
+        EXPECT_EQ( ghosted_shared_tile_space.min( Dim::J ) *
+                       cell_num_per_tile_dim,
+                   owned_cell_space.max( Dim::J ) );
+        EXPECT_EQ( ghosted_shared_tile_space.max( Dim::J ) *
+                       cell_num_per_tile_dim,
+                   owned_cell_space.max( Dim::J ) + halo_width );
+        EXPECT_EQ( ghosted_shared_tile_space.min( Dim::K ) *
+                       cell_num_per_tile_dim,
+                   owned_cell_space.min( Dim::K ) );
+        EXPECT_EQ( ghosted_shared_tile_space.max( Dim::K ) *
+                       cell_num_per_tile_dim,
+                   owned_cell_space.max( Dim::K ) );
+    }
+    if ( cart_rank[Dim::K] + 1 < cart_dims[Dim::K] )
+    {
+        auto ghosted_shared_tile_space =
+            local_grid_ptr->sharedTileIndexSpace<2>( Ghost(), Cell(), 0, 0, 1 );
+        EXPECT_EQ( ghosted_shared_tile_space.min( Dim::I ) *
+                       cell_num_per_tile_dim,
+                   owned_cell_space.min( Dim::I ) );
+        EXPECT_EQ( ghosted_shared_tile_space.max( Dim::I ) *
+                       cell_num_per_tile_dim,
+                   owned_cell_space.max( Dim::I ) );
+        EXPECT_EQ( ghosted_shared_tile_space.min( Dim::J ) *
+                       cell_num_per_tile_dim,
+                   owned_cell_space.min( Dim::J ) );
+        EXPECT_EQ( ghosted_shared_tile_space.max( Dim::J ) *
+                       cell_num_per_tile_dim,
+                   owned_cell_space.max( Dim::J ) );
+        EXPECT_EQ( ghosted_shared_tile_space.min( Dim::K ) *
+                       cell_num_per_tile_dim,
+                   owned_cell_space.max( Dim::K ) );
+        EXPECT_EQ( ghosted_shared_tile_space.max( Dim::K ) *
+                       cell_num_per_tile_dim,
+                   owned_cell_space.max( Dim::K ) + halo_width );
+    }
+    //////////////////
+    // NODE SPACES
```

Comment:
Just leaving a note to add before we  merge

Reply:
What do you mean by "leaving a note" here? What kind of notes are expected?

Reply:
template for node/cell